agent/consul/state/config_entry.go

```diff
@@ -92,10 +92,10 @@ func (s *Store) ConfigEntry(ws memdb.WatchSet, kind, name string) (uint64, struc
 	if err != nil {
 		return 0, nil, fmt.Errorf("failed config entry lookup: %s", err)
 	}
+	ws.Add(watchCh)
 	if existing == nil {
 		return idx, nil, nil
```

Comment:
We still return a `0` index here. I think in the past we've talked about making `maxIndexTxn` return min 1 globally and agree it's probably a good general solution but always chickened out because it effects lots of things (and breaks lots of tests that currently assert 0 initially arbitrarily) and we tend to only deal with these issues at points where we don't especially want to make bigger changes than we need to!

I'd be up for throwing a min 1 in here or the RPC though?

Reply:
For the sake of documenting this for the future, we already handle this case generically in `blockingQuery` and don't allow returning a 0 index.